Zend/zend_exceptions.c

```diff
@@ -711,25 +726,70 @@ static const zend_function_entry error_exception_functions[] = {
 void zend_register_default_exception(void) /* {{{ */
 {
 	zend_class_entry ce;
+	zend_property_info *prop;
 
-	INIT_CLASS_ENTRY(ce, "Exception", default_exception_functions);
-	default_exception_ce = zend_register_internal_class(&ce);
-	default_exception_ce->create_object = zend_default_exception_new;
+	INIT_CLASS_ENTRY(ce, "BaseException", default_exception_functions);
+	base_exception_ce = zend_register_internal_class(&ce);
+	base_exception_ce->ce_flags |= ZEND_ACC_EXPLICIT_ABSTRACT_CLASS;
+	base_exception_ce->create_object = NULL;
 	memcpy(&default_exception_handlers, zend_get_std_object_handlers(), sizeof(zend_object_handlers));
 	default_exception_handlers.clone_obj = NULL;
 
-	zend_declare_property_string(default_exception_ce, "message", sizeof("message")-1, "", ZEND_ACC_PROTECTED);
-	zend_declare_property_string(default_exception_ce, "string", sizeof("string")-1, "", ZEND_ACC_PRIVATE);
-	zend_declare_property_long(default_exception_ce, "code", sizeof("code")-1, 0, ZEND_ACC_PROTECTED);
-	zend_declare_property_null(default_exception_ce, "file", sizeof("file")-1, ZEND_ACC_PROTECTED);
-	zend_declare_property_null(default_exception_ce, "line", sizeof("line")-1, ZEND_ACC_PROTECTED);
-	zend_declare_property_null(default_exception_ce, "trace", sizeof("trace")-1, ZEND_ACC_PRIVATE);
-	zend_declare_property_null(default_exception_ce, "previous", sizeof("previous")-1, ZEND_ACC_PRIVATE);
+	zend_declare_property_string(base_exception_ce, "message", sizeof("message")-1, "", ZEND_ACC_PROTECTED);
+	zend_declare_property_string(base_exception_ce, "string", sizeof("string")-1, "", ZEND_ACC_PRIVATE);
+	zend_declare_property_long(base_exception_ce, "code", sizeof("code")-1, 0, ZEND_ACC_PROTECTED);
+	zend_declare_property_null(base_exception_ce, "file", sizeof("file")-1, ZEND_ACC_PROTECTED);
+	zend_declare_property_null(base_exception_ce, "line", sizeof("line")-1, ZEND_ACC_PROTECTED);
+	zend_declare_property_null(base_exception_ce, "trace", sizeof("trace")-1, ZEND_ACC_PRIVATE);
+	zend_declare_property_null(base_exception_ce, "previous", sizeof("previous")-1, ZEND_ACC_PRIVATE);
+
+	INIT_CLASS_ENTRY(ce, "Exception", NULL);
+	default_exception_ce = zend_register_internal_class_ex(&ce, base_exception_ce);
+	default_exception_ce->create_object = zend_default_exception_new;
+
+	/* A trick, to make visible prvate properties of BaseException */
+	ZEND_HASH_FOREACH_PTR(&default_exception_ce->properties_info, prop) {
+		if (prop->flags & ZEND_ACC_SHADOW) {
+			if (prop->name->len == sizeof("\0BaseException\0string")-1) {
+				prop->flags &= ~ZEND_ACC_SHADOW;
+				prop->flags |= ZEND_ACC_PRIVATE;
+				prop->ce = default_exception_ce;
+				zend_new_interned_string(
+					zend_string_init("\0Exception\0string", sizeof("\0Exception\0string")-1, 1));
```

Comment:
I removed this dead code.
We may remove this hack at all.
It affects a single test Zend/tests/bug63762.phpt